src/states/node.rs

```diff
@@ -516,6 +524,95 @@ impl Node {
         Ok(())
     }
 
+    fn get_section(&self, prefix: &Prefix<XorName>) -> Result<BTreeSet<XorName>, RoutingError> {
+        let section = self.peer_mgr
+            .routing_table()
+            .get_section(&prefix.lower_bound())
+            .ok_or(RoutingError::InvalidSource)?
+            .iter()
+            .cloned()
+            .collect();
+        Ok(section)
+    }
+
+    fn get_section_list(&self, prefix: &Prefix<XorName>) -> Result<GroupList, RoutingError> {
+        Ok(GroupList {
+            pub_ids: self.peer_mgr
+                .get_pub_ids(&self.get_section(prefix)?.into_iter().collect())
```

Comment:
That's because of a `HashMap` vs `BTreeMap` mismatch and me not wanting to duplicate the code. But I agree that there probably should be a better way.